### PR TITLE
Fix a bunch of errorlint errors from our linters

### PR DESCRIPTION
**Pull request by ph**
This fixes a lot of the error detecting by our linters, this skip the
errors related to the errors packages, we should refactor the usage or
our package instead.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,6 @@ notice:
 .PHONY: check-ci
 check-ci:
 	@mage update
-	@mage check
```
Comment (ph) [inline Makefile:47]: This shouldn't be here, this actually makes the linter run twice and it should only be run by the github action.

**Review by AndersonQ — CHANGES_REQUESTED** [review]
I think you might have misses a `!` on a if statement. I commented there

File: internal/pkg/agent/operation/operator.go [hunk 2]
```diff
@@ -164,15 +164,15 @@ func (o *Operator) Close() error {
 func (o *Operator) HandleConfig(ctx context.Context, cfg configrequest.Request) (err error) {
 	span, ctx := apm.StartSpan(ctx, "route", "app.internal")
 	defer func() {
-		if err = filterContextCancelled(err); err != nil {
+		if errors.Is(err, context.Canceled) {
```
[the remark this answers is not in the document]
Comment (ph) [reply internal/pkg/agent/operation/operator.go:167]: I think you are correct here, the filtering is non-obvious I was misled by it.

**Review by AndersonQ — APPROVED** [review]
you still have failing tests, but so far so good :)

**Comment by ph** [comment]
@AndersonQ well the linter... is a work in progress. I think the next one is the errcheck.
@mdelapenya or @ChrsMark I see an e2e test failing on k8s, Is this a known issue or I should investigate it?

```

[2022-04-26T14:24:10.545Z]     Then "elastic-agent" collects events with "kubernetes.pod.name:redis" # autodiscover_test.go:295 -> *podsManager

[2022-04-26T14:24:10.545Z]     context deadline exceeded

[2022-04-26T14:24:10.545Z] TRAC[2022-04-26T14:24:09Z] Executing command                             args="[export logs --name kind-a2627034-35d0-4f57-9620-83f7ceb14414 /home/admin/e2e-testing/outputs/kubernetes-autodiscover/kind-a2627034-35d0-4f57-9620-83f7ceb14414]" command=kind env="map[]"

[2022-04-26T14:24:10.546Z] TRAC[2022-04-26T14:24:09Z] Output                                        output=/home/admin/e2e-testing/outputs/kubernetes-autodiscover/kind-a2627034-35d0-4f57-9620-83f7ceb14414

[2022-04-26T14:24:10.546Z] INFO[2022-04-26T14:24:09Z] Kind cluster logs exported                    cluster=kind-a2627034-35d0-4f57-9620-83f7ceb14414 path=/home/admin/e2e-testing/outputs/kubernetes-autodiscover/kind-a2627034-35d0-4f57-9620-83f7ceb14414

[2022-04-26T14:24:10.546Z] TRAC[2022-04-26T14:24:09Z] Executing command                             args="[delete cluster --name kind-a2627034-35d0-4f57-9620-83f7ceb14414]" command=kind env="map[]"

[2022-04-26T14:24:11.932Z] TRAC[2022-04-26T14:24:11Z] Output                                        output=

[2022-04-26T14:24:11.932Z] INFO[2022-04-26T14:24:11Z] kind cluster  was deleted                    

[2022-04-26T14:25:19.687Z] 

[2022-04-26T14:25:19.687Z] --- Failed steps:

[2022-04-26T14:25:19.687Z] 

[2022-04-26T14:25:19.687Z]   Scenario: Logs collection from running pod # features/elastic_agent_standalone.feature:9

[2022-04-26T14:25:19.687Z]     Then "elastic-agent" collects events with "kubernetes.pod.name:a-pod" # features/elastic_agent_standalone.feature:12

[2022-04-26T14:25:19.687Z]       Error: context deadline exceeded

[2022-04-26T14:25:19.687Z] 

[2022-04-26T14:25:19.687Z]   Scenario: Logs collection from a pod with an init container # features/elastic_agent_standalone.feature:14

[2022-04-26T14:25:19.687Z]     Then "elastic-agent" collects events with "kubernetes.container.name:init-container" # features/elastic_agent_standalone.feature:17

[2022-04-26T14:25:19.687Z]       Error: context deadline exceeded

[2022-04-26T14:25:19.687Z] 

[2022-04-26T14:25:19.687Z]   Scenario: Logs collection from short-living cronjobs # features/elastic_agent_standalone.feature:22

[2022-04-26T14:25:19.687Z]     Then "elastic-agent" collects events with "kubernetes.container.name:cronjob-container" # features/elastic_agent_standalone.feature:26

[2022-04-26T14:25:19.687Z]       Error: context deadline exceeded

[2022-04-26T14:25:19.687Z] 

[2022-04-26T14:25:19.687Z]   Scenario: Logs collection from failing pod # features/elastic_agent_standalone.feature:28

[2022-04-26T14:25:19.687Z]     Then "elastic-agent" collects events with "kubernetes.pod.name:a-failing-pod" # features/elastic_agent_standalone.feature:31

[2022-04-26T14:25:19.687Z]       Error: context deadline exceeded

[2022-04-26T14:25:19.687Z] 

[2022-04-26T14:25:19.687Z]   Scenario: Metrics collection configured from targeted Redis Pod # features/elastic_agent_standalone.feature:33

[2022-04-26T14:25:19.687Z]     Then "elastic-agent" collects events with "kubernetes.pod.name:redis" # features/elastic_agent_standalone.feature:36

[2022-04-26T14:25:19.687Z]       Error: context deadline exceeded

[2022-04-26T14:25:19.687Z] 
```

**Comment by mdelapenya** [comment]
> @AndersonQ well the linter... is a work in progress. I think the next one is the errcheck. @mdelapenya or @ChrsMark I see an e2e test failing on k8s, Is this a known issue or I should investigate it?

The team is working on it, thanks for the ping!

**Pull request merged**